$NetBSD: patch-libterm_tgoto_c,v 1.1 2012/12/28 03:03:09 dholland Exp $

- use const for string constants
- silence warnings about assignments in conditionals

--- libterm/tgoto.c~	2002-02-04 01:30:04.000000000 +0000
+++ libterm/tgoto.c
@@ -83,7 +83,7 @@ char	*BC;
  *
  * all other characters are ``self-inserting''.
  */
-char *
+const char *
 tgoto(CM, destcol, destline)
 	char *CM;
 	int destcol, destline;
@@ -104,7 +104,7 @@ toohard:
 		return ("OOPS");
 	}
 	added[0] = 0;
-	while (c = *cp++) {
+	while ((c = *cp++) != 0) {
 		if (c != '%') {
 			*dp++ = c;
 			continue;
